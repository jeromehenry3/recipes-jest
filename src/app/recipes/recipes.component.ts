import { Component, OnInit } from '@angular/core';

@Component({
  selector: 'app-recipes',
  templateUrl: './recipes.component.html',
  styleUrls: ['./recipes.component.scss']
})
export class RecipesComponent implements OnInit {

  constructor() { }

  ngOnInit(): void {
  }

  fakeFuncTocheckCoverageIsOk() {
    let a = 1;
    a = 2;
    const c = 3;
    console.log('c', c);
    return a + c;
  }

}
