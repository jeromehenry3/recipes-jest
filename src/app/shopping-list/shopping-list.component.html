<div class="row">
  <div class="col-xs-10">
    <app-shopping-edit></app-shopping-edit>
    <hr />
    <p>ici vient la liste</p>
  </div>
</div>
