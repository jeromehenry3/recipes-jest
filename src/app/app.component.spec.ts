import { TestBed, async } from '@angular/core/testing';
import { AppComponent } from './app.component';
import { HeaderComponent } from './header/header.component';
import { RecipesComponent } from './recipes/recipes.component';
import { By } from '@angular/platform-browser';
import { ShoppingListComponent } from './shopping-list/shopping-list.component';

describe('AppComponent', () => {
  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
        AppComponent,
        HeaderComponent,
        RecipesComponent,
        ShoppingListComponent,
      ],
      providers: [
        HeaderComponent,
        RecipesComponent,
        ShoppingListComponent,
      ]
    }).compileComponents();
  }));

  it('should create the app', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.componentInstance;
    expect(app).toBeTruthy();
  });

  it(`should have as title 'recipes-jest'`, () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.componentInstance;
    expect(app.title).toEqual('recipes-jest');
  });

  it('should match snapshot', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.debugElement.nativeElement;
    expect(app).toMatchSnapshot();
  })

  describe('App should render its child components', () => {
    beforeEach(async(() => {
      TestBed.configureTestingModule({
        declarations: [
          AppComponent,
          HeaderComponent,
          RecipesComponent,
        ],
        providers: [
          HeaderComponent,
        ]
      }).compileComponents();
    }));
    it('should render the `Header` component', () => {
      const fixture = TestBed.createComponent(AppComponent);
      const app = fixture.debugElement;
      fixture.detectChanges();
      const header = app.query(By.directive(HeaderComponent));
      expect(header.componentInstance).toBeInstanceOf(HeaderComponent);
    });
    it('should render the `Recipes` component', () => {
      const fixture = TestBed.createComponent(AppComponent);
      const app = fixture.debugElement;
      fixture.detectChanges();
      const recipes = app.query(By.directive(RecipesComponent));
      expect(recipes.componentInstance).toBeInstanceOf(RecipesComponent);
    });
    it('should render the `ShoppingLIst` component', () => {
      const fixture = TestBed.createComponent(AppComponent);
      const app = fixture.debugElement;
      fixture.detectChanges();
      const recipes = app.query(By.directive(ShoppingListComponent));
      expect(recipes.componentInstance).toBeInstanceOf(ShoppingListComponent);
    });
  });

});
