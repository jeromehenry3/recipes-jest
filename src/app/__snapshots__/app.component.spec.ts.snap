// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should match snapshot 1`] = `
<div
  id="root2"
>
  <app-header>
    <p>
      header works!
    </p>
  </app-header>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="col-md-12"
      >
        <app-recipes>
          <p>
            recipes works!
          </p>
        </app-recipes>
        <app-shopping-list>
          <p>
            shopping-list works!
          </p>
        </app-shopping-list>
      </div>
    </div>
  </div>
</div>
`;
